google/cloud/spanner/client.cc

```diff
@@ -135,6 +135,74 @@ StatusOr<BatchDmlResult> Client::ExecuteBatchDml(
       {std::move(transaction), std::move(statements)});
 }
 
+StatusOr<CommitResult> Client::Commit(
+    std::function<StatusOr<Mutations>(Transaction)> const& mutator,
+    std::unique_ptr<TransactionRerunPolicy> rerun_policy,
+    std::unique_ptr<BackoffPolicy> backoff_policy) {
+  // The status-code discriminator of TransactionRerunPolicy.
+  using RerunnablePolicy = internal::SafeTransactionRerun;
+
+  for (int rerun = 0;; ++rerun) {
+    // TODO(#472): Make this transaction use the same session each time.
+    Transaction txn = MakeReadWriteTransaction();
+    StatusOr<Mutations> mutations;
```

Comment:
Does this need to be declared here, or can it be declared on line 152 when it's initialized?

Reply:
I'll say yes, it does need to be declared here, otherwise it would (`#if`-ly) be out-of-scope on lines 163 and beyond.